browser/ui/views/download/brave_download_item_view.cc

```diff
@@ -99,7 +100,9 @@ void BraveDownloadItemView::OnPaint(gfx::Canvas* canvas) {
   DownloadItemView::OnPaint(canvas);
   if (is_origin_url_visible_)
     DrawOriginURL(canvas);
-  file_name_label_->SetVisible(!is_origin_url_visible_);
+  file_name_label_->SetVisible(
+      !is_origin_url_visible_ &&
+      (GetMode() == download::DownloadItemMode::kNormal));
```

Comment:
`OnPaint` seems an unusual place to change visibility.

Reply:
should this ideally be abstracted to separate function `DrawFilenameLabel` and that's where the visibility can be set?

Reply:
I think we should be changing the `file_name_label_` visibility at the same time we are changing the `is_origin_url_visible_` value, as I think currently we are just painting over the file name with the origin url.
Maybe add a private method:
<pre>
void SetOriginUrlVisible(bool visible) {
  is_origin_url_visible_ = visible;
  file_name_label_->SetVisible(!visible);
}
</pre>
and use it everywhere we currently set `is_origin_url_visible_` directly. 

But I am also not clear why we are seeing that drawing artifact when we aren't currently deciding when the `file_name_label_` is visible. Is this happening in Chrome as well?

Reply:
Good call @mkarolin - I'm only seeing Chromium set inside the `UpdateLabels` method. Comment says: `Updates the visibility, text, size, etc. of all labels.`. We could potentially override this too
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/views/download/download_item_view.cc;l=812;drc=81c573db62422920e3f2fb3edef0226b7aff17f6

Reply:
yep, that artifact gets rendered because upstream sets it inside `UpdateLabels`

Reply:
I have moved the logic from `OnPaint` to appropriate the function.